File: sdks/java/examples/CredentialVerification.java
import dev.atap.sdk.ATAPClient;
import dev.atap.sdk.model.Credential;

import java.util.List;

/**
 * Example: Email verification and credential management.
 */
public class CredentialVerification {

    public static void main(String[] args) {
        try (ATAPClient client = ATAPClient.builder()
                .baseUrl("http://localhost:8080")
                .did("did:web:localhost%3A8080:human:user123")
                .privateKey("<base64-encoded-private-key>")
                .clientSecret("atap_<your-secret>")
                .build()) {

            // 1. Start email verification
            String message = client.credentials.startEmail("user@example.com");
            System.out.println(message); // "OTP sent to user@example.com"

            // 2. Verify with OTP (received via email)
            Credential emailCred = client.credentials.verifyEmail("user@example.com", "123456");
            System.out.println("Email credential: " + emailCred.getId());
            System.out.println("Type: " + emailCred.getType());
            System.out.println("VC JWT: " + emailCred.getCredential());

            // 3. Start phone verification
            client.credentials.startPhone("+15551234567");

            // 4. Verify with OTP
            Credential phoneCred = client.credentials.verifyPhone("+15551234567", "654321");
            System.out.println("Phone credential: " + phoneCred.getId());

            // 5. Submit personhood attestation
            Credential personhood = client.credentials.submitPersonhood();
            System.out.println("Personhood credential: " + personhood.getId());

            // 6. List all credentials
            List<Credential> allCreds = client.credentials.list();
            System.out.println("Total credentials: " + allCreds.size());
        }
    }
}

File: sdks/php/examples/credential_verification.php
<?php

/**
 * Example: Email/phone/personhood credential verification flow.
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Atap\Sdk\ATAPClient;

$client = new ATAPClient([
    'base_url' => 'http://localhost:8080',
    'did' => 'did:web:localhost%3A8080:human:user123',
    'private_key' => '<base64-encoded-ed25519-seed>',
    'client_secret' => 'atap_...',
]);

// 1. Start email verification
echo "Starting email verification...\n";
$message = $client->credentials()->startEmail('user@example.com');
echo "  {$message}\n";

// 2. Verify email with OTP
$emailCred = $client->credentials()->verifyEmail('user@example.com', '123456');
echo "Email credential issued: {$emailCred->id}\n";
echo "  Type: {$emailCred->type}\n";

// 3. Start phone verification
echo "\nStarting phone verification...\n";
$message = $client->credentials()->startPhone('+1234567890');
echo "  {$message}\n";

// 4. Verify phone with OTP
$phoneCred = $client->credentials()->verifyPhone('+1234567890', '654321');
echo "Phone credential issued: {$phoneCred->id}\n";

// 5. Submit personhood attestation
echo "\nSubmitting personhood attestation...\n";
$personhood = $client->credentials()->submitPersonhood();
echo "Personhood credential issued: {$personhood->id}\n";

// 6. List all credentials
$credentials = $client->credentials()->list();
echo "\nAll credentials ({$credentials} total):\n";
foreach ($credentials as $cred) {
    echo "  - {$cred->type} (issued: {$cred->issuedAt})\n";
}

// 7. Check public status list
$statusList = $client->credentials()->statusList('1');
echo "\nStatus list type: " . implode(', ', $statusList['type'] ?? []) . "\n";
